### ISSUE TITLE
NPE in SentryFeignClient

### ISSUE THREAD

**Issue by neseleznev** [getsentry/sentry-java#1907]
_Platform:_
- [x]  Java -> If yes, which Java (and sourceCompatibility/targetCompatibility) version? **11/11**
- [x]  Spring -> If yes, which version? **Spring Boot 2.4.9 / Spring 5.3.9**

_Platform installed with:_
- [x] Maven Central

The version of the SDK:
**5.6.1** (the newest with my previous fix rolled out)

-----

I have the following issue:
This is another bugfix to the issue #1883

NPE we caught two weeks ago raised again. Take a look
```
java.lang.NullPointerException: null
	at io.sentry.openfeign.SentryFeignClient.addBreadcrumb(SentryFeignClient.java:89) ~[sentry-openfeign-5.6.1.jar!/:na]
	at io.sentry.openfeign.SentryFeignClient.execute(SentryFeignClient.java:76) ~[sentry-openfeign-5.6.1.jar!/:na]
	at feign.SynchronousMethodHandler.executeAndDecode(SynchronousMethodHandler.java:119) ~[feign-core-10.12.jar!/:na]
	at feign.SynchronousMethodHandler.invoke(SynchronousMethodHandler.java:89) ~[feign-core-10.12.jar!/:na]
	at feign.ReflectiveFeign$FeignInvocationHandler.invoke(ReflectiveFeign.java:100) ~[feign-core-10.12.jar!/:na]
	at com.sun.proxy.$Proxy172.findAccountOfCurrentUser(Unknown Source) ~[na:na]
	at 
```
Where line 89 is previously fixed statement
```java
    if (response != null) {
      breadcrumb.setData(
          "response_body_size", response.body() != null ? response.body().length() : 0);  // <<< line 89
```

Unfortunately, we don't have pretty NPE explanation from Java 14+ since we still use JDK 11.
However, ternary unboxing behavior is notoriously known, see https://stackoverflow.com/a/33182091
Thus, null `response.body().length()` will cause NPE straight in that line of code.

I dug deeper into OpenFeign, where it's clearly stated that null length is possible.
```java
/**
     * length in bytes, if known. Null if unknown or greater than {@link Integer#MAX_VALUE}.
     *
     * <br>
     * <br>
     * <br>
     * <b>Note</b><br>
     * This is an integer as most implementations cannot do bodies greater than 2GB.
     */
    Integer length();
```

In particular, InputStreamBody has `Integer length` set in constructor, which may be unknown, and `null` https://github.com/OpenFeign/feign/blob/f71849d38746fdc7871bf3b0609c488b36459687/core/src/main/java/feign/Response.java#L295

TL;DR

I'm really sorry I couldn't fix it last time, which is complicated by the fact that the NPE rose only in `qa` & `prod` environments.
I rolled out the final fix in PR

**Comment by marandaneto** [comment]
@neseleznev thanks for fixing this.
